src/runtime/free-access.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FreeAccessApi} from '../api/free-access-api';
+import {LocationProvider} from './location-provider';
+
+export class FreeAccess implements FreeAccessApi {
+  private static readonly GOOGLE_REFERERS_PREFIX_LIST = [
+    'https://www.google.com/',
```

Comment:
Wondering if the rules are too strict. Is it possible that we get http referrals or referrals without a trailing `/`?